TESTPLAN.md

```diff
@@ -163,72 +85,56 @@ All UI `shield-study` `study_state` sequences look like this:
 
       - ending is `window-or-fx-closed`
 
+5.  UI functionality  'too-popular'
 
----
-
-## Helper code and tips
-
-### ***To open a Chrome privileged console***
-
-1.  `about:addons`
-2.  `Tools > web developer console`
-
-Or use other methods, like Scratchpad.
-
-
-### **Telemetry Ping Printing Helper Code**
-
-```javascript
-async function printPings() {
-  async function getTelemetryPings (options) {
-    // type is String or Array
-    const {type, n, timestamp, headersOnly} = options;
-    Components.utils.import("resource://gre/modules/TelemetryArchive.jsm");
-    // {type, id, timestampCreated}
-    let pings = await TelemetryArchive.promiseArchivedPingList();
-    if (type) {
-      if (!(type instanceof Array)) {
-        type = [type];  // Array-ify if it's a string
-      }
-    }
-    if (type) pings = pings.filter(p => type.includes(p.type));
-    if (timestamp) pings = pings.filter(p => p.timestampCreated > timestamp);
-
-    pings.sort((a, b) => b.timestampCreated - a.timestampCreated);
-    if (n) pings = pings.slice(0, n);
-    const pingData = headersOnly ? pings : pings.map(ping => TelemetryArchive.promiseArchivedPingById(ping.id));
-    return Promise.all(pingData);
-  }
-  async function getPings() {
-    const ar = ["shield-study", "shield-study-addon"];
-    return getTelemetryPings({type: ar});
-  }
-
-  const pings = (await getPings()).reverse();
-  const p0 = pings[0].payload;
-  // print common fields
-  console.log(
-    `
-// common fields
-
-branch        ${p0.branch}        // should describe Question text
-study_name    ${p0.study_name}
-addon_version ${p0.addon_version}
-version       ${p0.version}
-
-    `
-  );
-
-  pings.forEach(p => {
-    console.log(p.creationDate, p.payload.type);
-    console.log(JSON.stringify(p.payload.data, null, 2));
-  });
-}
-
-printPings();
-```
+    * Click on the web extension's icon three times
+    * Verify that the study ends
+    * Verify that sent Telemetry is correct
+    * Verify that the user is sent to the URL specified in `addon/Config.jsm` under `endings -> too-popular`.
+
+## Automated Testing
+
+`npm run test` verifies the telemetry payload as expected at Firefox startup and add-on installation in a clean profile, then does **optimistic testing** of the *commonest path* though the study for a user
+
+- prove the notification bar ui opens
+- *clicking on the left-most button presented*.
+- verifying that sent Telemetry is correct.
+
+Code at [/test/functional_test.js](/test/functional_test.js).
+
+### Note: checking "sent Telemetry is correct"
+
+* Open the Browser Console using Firefox's top menu at `Tools > Web Developer > Browser Console`. This will display Shield (loading/telemetry) log output from the add-on.
```

Comment:
Perhaps we can reference TELEMETRY.md and/or TESTPLAN.md here for how to verify "Telemetry is correct", since that's a nebulous phrase? In general I think we can expand on this, but that can be a separate PR later.